### ISSUE TITLE
False positive with short variable declaration operator

### ISSUE THREAD

**Issue by Kantonov** [golangci/golangci-lint#2834]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

The issue came with the latest version, 1.45.2 didn't report that. I've added a simple code example. If the `a := decimal.Zero` is replaced with `var a = decimal.Zero` everything is cool.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.46.0 built from 7c1d8b75 on 2022-05-09T11:16:24Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
no config 
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.16.6 darwin/amd64
GO111MODULE=""
GOARCH="amd64"
GOVERSION="go1.16.6"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOGCCFLAGS="-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/t8/5brmb_nd5cs5v4w_vzxyhft40000gn/T/go-build512567077=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [runner] linters took 44.176085ms with stages: goanalysis_metalinter: 43.622345ms
main.go:9:2: SA4006: this value of `a` is never used (staticcheck)
	a := decimal.Zero
	^
```

</details>

### Code example or link to a public repository

<details>

```go
package lint_test

import (
	"github.com/shopspring/decimal"
	"time"
)

func Main1() {
	a := decimal.Zero
	var b = decimal.NewFromInt(137)
	if time.Now().Unix() > 137 {
		a = b
		var c = MyType{}
		c.DoStuff(a.String())
	}
}

type MyType struct{}

func (mt MyType) DoStuff(s string) {}
```

</details>

**Comment by ldez** [comment]
Hello,

have you tried to use staticcheck as a standalone command?

https://github.com/dominikh/go-tools

**Comment by ldez** [comment]
I take a look at this issue, and in fact, in your example the linter is right: `a` is not used because you only use `b`.

If I rewrite your code, you can see that only `b` is used:
```go
func Main1() {
	// a := decimal.Zero
	var b = decimal.NewFromInt(137)
	if time.Now().Unix() > 137 {
		// a = b
		var c = MyType{}
		// c.DoStuff(a.String())
		c.DoStuff(b.String())
	}
}
```
